bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/embed/OEmbedClientImplConfigurationFactory.java

```diff
@@ -77,6 +77,18 @@ public class OEmbedClientImplConfigurationFactory {
             description = "Describes whether the provider response HTML is allowed to be displayed in an unsafe context."
         )
         boolean unsafeContext() default false;
+
+        @AttributeDefinition(
+            name = "Socket Timeout",
+            description = "The time waiting for data – after establishing the connection; maximum time of inactivity between two data packets."
+        )
+        int socketTimeout() default 1000;
```

Comment:
Should we use these aggressive timeouts probably proposed by the customer as default?

For me the safer way would be to preserve the old defaults for both timeouts and let this customer configure the more aggressive timeouts for their solution. Such dramatic change of the defaults (60x and 5x reduction) may cause surprises for other customers who weren't complaining about this.

Reply:
In my opinion we should use an aggressive timeout by default that can be changed by the customers based on the providers they use. In this case the customers that are using our default providers don't have to wait unnecessary time for connection. We have defined config for providers that are offering a fast connection. 
As a middle ground, could we set a less aggressive timeout such 3000ms that should cover both opinions?

Reply:
@andreeadracea , I've looked into the consequences of using too aggressive timeouts. The way the apache HTTP connection is used implies if a network error occurs (no route to hots, connection timeout, socket timeout etc) then the edit dialog of embed component cannot validate the url and during page rendering for endusers a page with embed component cannot render the embedded player. There is not any kind of retry in place, the only option is for the user to reload the page or the author try to enter the url again in the edit dialog. Without retry a very aggressive timeout is not recommended but the connection between the AEM instance and the embedded service must be fast so, I also think we can reduce the default timeouts. My suggestion is to use 2000 ms for connection timeout and 5000 ms for socket timeout.

@vladbailescu , @joerghoh what do you think?